sdk/appconfiguration/tests.yml

```diff
@@ -6,6 +6,8 @@ jobs:
       ServiceDirectory: appconfiguration
       EnvVars:
         AZURE_TEST_MODE: RECORD
+        AZURE_TEST_HTTP_CLIENTS: ALL
+        AZURE_APPCONFIG_TEST_SERVICE_VERSIONS: ALL
```

Comment:
Mind providing some background or explanation in the PR comments on what this and how it works? Based on the PR alone I'm assuming there is some internal pluming that handles running these permutations but it is not abundantly clear on what based on the code changes.

Reply:
@alzimmermsft the plumbing to support this happened in this PR - https://github.com/Azure/azure-sdk-for-java/pull/8679

Reply:
Updated the description. Thanks for comments.